### PR TITLE
Initial pr

### PR DESCRIPTION
**Pull request by raghav-deepsource**
Adds a small demo java project using Gradle. It demonstrates a multimodule gradle repo with some errors in it.

### CODE REVIEW

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        this.configs = configs;
+    }
+
+    /**
+     * Shortcut for calling wait
+     */
+    private void waitForLock(Condition c) {
+        try {
+            c.wait();
+        } catch (Throwable e) {}
```
Comment (sourya-deepsource) [inline multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java:46]: Is this block intentionally left empty?

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public void getDataInParallel() throws InterruptedException {
+
+        Thread[] ts = new Thread[configs.length];
+
+        // Locks make use of condition variables for synchronization.
+        Condition prevDone = LOCK.newCondition();
+        for (int i = 0; i < configs.length; ++i) {
+            int finalI = i;
+            ts[i] = new Thread(
+                    () -> {
```
Comment (sourya-deepsource) [inline multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java:63]: Is this a valid pattern in Java, or is it meant to raise issues?

File: source/com/example/Main.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * lorem ipsum sit dor tass
+ */
+public class Main {
+
+    static ArrayList<ConfigData> configs;
+
+    /**
+     * Documented function
+     *
+     * @param args adb, ddhdhfwdfe
```
Comment (sourya-deepsource) [inline source/com/example/Main.java]: Can the second arg be named better?

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java (hunk 1 above)
Comment (raghav-deepsource) [reply multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java:46]: yeah

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java (hunk 2 above)
Comment (raghav-deepsource) [reply multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java:63]: it was an ide suggestion

File: source/com/example/Main.java (hunk 3 above)
Comment (raghav-deepsource) [reply source/com/example/Main.java]: yup

**Review by sourya-deepsource — DISMISSED** [review]

File: gradle.properties [hunk 4]
```diff
@@ -1,4 +1,14 @@
 
+org.gradle.jvmargs=-Xmx2024m -XX:MaxPermSize=512m
```
Comment (sourya-deepsource) [inline gradle.properties]: Why does this have so many repeating lines?

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java [hunk 5]
```diff
@@ -7,6 +7,9 @@
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
+/**
+ * Performs a network call for
```
Comment (sourya-deepsource) [inline multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java]: For?

File: gradle.properties (hunk 4 above)
Comment (raghav-deepsource) [reply gradle.properties]: its a bug in the analyzer that only manifests itself when the analyzer runs multiple times on the same repo in the same run

File: multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java (hunk 5 above)
Comment (raghav-deepsource) [reply multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java]: argh fixed now

File: gradle.properties (hunk 4 above)
Comment (raghav-deepsource) [reply gradle.properties]: basically only happens locally coz I run analysis multiple times and some changes accumulate.

Comment (sourya-deepsource) [reply gradle.properties]: Local analysis?

Comment (raghav-deepsource) [reply gradle.properties]: it doesn't cause trouble on dev or prod tho.

Comment (raghav-deepsource) [reply gradle.properties]: yes, local analysis

Comment (sourya-deepsource) [reply gradle.properties]: Cool

**Review by sourya-deepsource — APPROVED** [review]

**Review by souvik-deepsource — APPROVED** [review]

**Pull request merged**